### ISSUE TITLE
Awk has a syntax error in the benchmarks

### ISSUE THREAD

**Issue by rempas** [vnmakarov/mir#382]
Specifically, the line shows: `awk: cmd. line:1: BEGIN {if (2,88==0) print "Inf"; else printf "%.2f\n", 2,88/2,88;}
awk: cmd. line:1:             ^ syntax error`.

**Comment by rempas** [comment]
> It works for me on Linux. It seems `2,88` is taken from some time utility which outputs time with comma. Probably LC_NUMERIC was set up on your system.
> 
> Could you give more info how to reproduce it.

Well... I truly haven't changed anything! I could give you info about my system!

Linux: `6.6.2.arch1-1`
Awk: `GNU Awk 5.3.0, API 4.0, PMA Avon 8-g1, (GNU MPFR 4.2.1, GNU MP 6.3.0)`

**Comment by iacore** [comment]
Have you tried removing all `LC*` and `LANG*` environment variables before running the benchmark?

**Comment by rempas** [comment]
> Have you tried removing all `LC*` and `LANG*` environment variables before running the benchmark?

@vnmakarov said the same, but I'm apparently blind :facepalm: 

Yes, "LC_NUMERIC" is set. Unsetting it solves the problems.

Now, before I close that. Should the script unset it manually?

**Comment by iacore** [comment]
> Now, before I close that. Should the script unset it manually?

It should, although that's a work around.


What was your `LC_NUMERIC` set to?

**Comment by iacore** [comment]
I cannot reproduce this.

What I ran in fish:

```
set -x LC_NUMERIC el_GR.UTF-8
make --debug bench
```

what command did you use to run the benchmark?

**Comment by rempas** [comment]
> I cannot reproduce this.
> 
> What I ran in fish:
> 
> ```
> set -x LC_NUMERIC el_GR.UTF-8
> make --debug bench
> ```
> 
> what command did you use to run the benchmark?

I am using ZSH (kernel and awk info above) and I run: `./c-benchmarks/run-benchmarks.sh`

**Comment by rempas** [comment]
> Thank you for the discussion. I addrd clearing LC_NUMERIC to the script.

Thanks. Even if @iacore didn't have the same problem (when trying to set LC_NUMERIC) and some others may not, you will save some time from the issues of people who will have that problem like me.

> If you use master branch, you should check `bbv` branch. It generates a better code (by 5-10%). Also, does this change the compilation times?
 
Thanks for the info. We have talked about it, and I was using that branch for my test until now, but I thought about sticking to `master` as these two will probably merge one day, right? If not, what is `bbv` about, and should I always use this, especially when testing?

> although the data are not fresh.

Thank you for the info, but 2 and a half years ago is a loooooot of time! In that period, clang (llvm) went from both compiling slower and producing worse code than GCC to basically doing an 180 degree and beating GCC on my system!

So yeah, I prefer to have the benchmark suit as a good indication of how things are going as compilers change rapidly. The only thing I would suggest, is to add compile times for every test separately and a sum in the end.

In any case, your work opens new possibility and ways to think about compilation! I have tried to compile [this](https://qawithexperts.com/article/c-cpp/fibonacci-series-program-in-c-with-and-without-recursion/300) program (no recursion version and slightly modified) and the output file is **16.000** bytes for clang and just **360** bytes for Mir!!! While the execution takes more for Mir. And I suppose you can still do a lot of optimization for both files sizes, compilation speed and code quality compared to LLVM who is already optimized as much as possible (or at least "practically" possible). JIT is the future!!!!

I will try with the `bbv` branch and come I'll post update!

**Comment by iacore** [comment]
> I would like help making the benchmark tool more advanced, but as I don't know shell scripting, I have another idea. I will create another benchmark suite written in Typescript, and it will require [bun](https://github.com/oven-sh/bun) to run.

If you want, try Janet. you can include the language itself as `janet.c` and bootstrap it from C compiler in source.

**Comment by iacore** [comment]
I don't think languages other than C or shell script fit in this project though.

**Comment by rempas** [comment]
> I don't think languages other than C or shell script fit in this project though. JS/bun is probably not a good fit.

Yeah, I would agree if we were talking about the source code, but this is about benchmarking, so I don't think that it's a big problem as long as it as SECOND choice and the shell script remains. That's why I didn't say to remove the current benchmarks.

> My reasoning was, Janet script can be compiled to C. If you know C, C is better.

I understand but like I said I don't have so much time. I know C but C will make my life so much harder...
